snuba/query/processors/physical/clickhouse_settings_override.py

```diff
@@ -0,0 +1,17 @@
+from typing import Any, MutableMapping
+
+from snuba.clickhouse.query import Query
+from snuba.query.processors.physical import ClickhouseQueryProcessor
+from snuba.query.query_settings import QuerySettings
+
+
+class ClickhouseSettingsOverride(ClickhouseQueryProcessor):
+    """
+    Overrides arbitrary clickhouse settings via a dictionary specifying the clickhouse settings to override.
+    """
+
+    def __init__(self, settings: MutableMapping[str, Any]) -> None:
+        self.__settings = settings
+
+    def process_query(self, query: Query, query_settings: QuerySettings) -> None:
+        query_settings.set_clickhouse_settings(self.__settings)
```

Comment:
Can we restrict the query settings which are modifiable by this processor? Snuba should have control over which settings can be modified rather than product teams.

Also, this should not be made generic. We should put additional safeguards in place like known referrers etc. on which we allow clickhouse setting override. That would allow better control over clickhouse usage and narrowing down bad referrers who might abuse clickhouse with some sort of query settings

Reply:
Nikhar and I spoke offline. This query processor is no different to changing the clickhouse configuration for the user, but easier to manage and revert if need be than an ops repo change. Initially there was some confusion which made it seem like we were accepting arbitrary query settings from the API user (which we are not)